### PR TITLE
[TorchFX][FBC] Constant linear layers support

### PR DESCRIPTION
**Pull request by daniil-lyakhov**
### Changes

Constant linear layers support

### Reason for changes

To support swint_v2_s FBC

### Related tickets

#2766 

### Tests
Build post_training_quantization/444/ is finished successfully
Unit test `test_model_transformer.test_model_extraction` is presented

### CODE REVIEW

**Comment by daniil-lyakhov** [comment]
@AlexanderDokuchaev, tests should be arrived in https://github.com/openvinotoolkit/nncf/pull/2867. Tested at least by resnet18 sanity test for now

File: nncf/experimental/torch/fx/model_transformer.py [hunk 1]
```diff
@@ -97,7 +98,23 @@ def _apply_model_extraction(
         # TODO(dlyakhov): reduce memory consumption by
         # more optimal splitting implementation.
         splitted_gm = split_by_tags(model, tags)
-        return splitted_gm.extracted
+
+        extracted_model = splitted_gm.extracted
```
Comment (AlexanderDokuchaev) [inline nncf/experimental/torch/fx/model_transformer.py:102]: Please, add test case that will cover new case and a case with conv too, like 
```
def test_extracted_model(syntetic_model, extracted_ref):
...
```

**Review by AlexanderDokuchaev — CHANGES_REQUESTED** [review]

File: nncf/experimental/torch/fx/model_transformer.py (hunk 1 above)
Comment (daniil-lyakhov) [reply nncf/experimental/torch/fx/model_transformer.py:102]: This test is expected to be in a separate PR. swin_v2_s have this problematic const linear inside, is presented in the conformance test. Should it be enough for merge to experimental dir??

Comment (daniil-lyakhov) [reply nncf/experimental/torch/fx/model_transformer.py:102]: Unit test is presented, please check

File: tests/torch/test_models/synthetic.py [hunk 2]
```diff
@@ -501,3 +501,24 @@ def forward(self, x):
         unbinded_processed[0] = self.conv4(y_unbinded[0])
         y = torch.cat(unbinded_processed, axis=0)
         return y
+
+
+class ModelExtractionModel(torch.nn.Module):
```
Comment (AlexanderDokuchaev) [inline tests/torch/test_models/synthetic.py:506]: Name of classes does not reflect the modules.
And looks like it should be moved to `tests/torch/fx/...`

**Review by AlexanderDokuchaev — CHANGES_REQUESTED** [review]

File: nncf/experimental/torch/fx/model_transformer.py [hunk 3]
```diff
@@ -97,7 +98,23 @@ def _apply_model_extraction(
...
+        graph: torch.fx.Graph = extracted_model.graph
+        # Check extracted model has inputs.
+        # It is possible to have two constant inputs
+        # for a linear layer, an placeholder is being
```
[the remark this answers is not in the document]
Comment (daniil-lyakhov) [reply nncf/experimental/torch/fx/model_transformer.py:106]: No, but linear is an example in this comment

File: tests/torch/test_models/synthetic.py (hunk 2 above)
Comment (daniil-lyakhov) [reply tests/torch/test_models/synthetic.py:506]: This is an instance of `torch.nn.Module`, I see no contradictions here. There are several models named after a test which is using this model (ModelForGraphBuildingTest and etc). I would keep it here until it is necessary

File: tests/torch/fx/test_model_transformer.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+MODEL_EXTRACTION_CASES = (
+    ModelExtractionTestCase(ModelExtractionModel, (1, 1, 3, 3), PTModelExtractionCommand(["conv2d"], ["conv2d"])),
+    ModelExtractionTestCase(
+        ConstantModelExtractionModel, (1, 1, 3, 3), PTModelExtractionCommand(["conv2d"], ["conv2d"])
+    ),
+)
+
+
+@pytest.mark.parametrize("test_case", MODEL_EXTRACTION_CASES)
```
[the remark this answers is not in the document]
Comment (daniil-lyakhov) [reply tests/torch/fx/test_model_transformer.py]: Done

File: tests/torch/fx/test_model_transformer.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+@dataclass
+class ModelExtractionTestCase:
+    model: torch.nn.Module
+    input_shape: Tuple[int, ...]
+    command: PTModelExtractionCommand
+    ref: None = None
+
+
+EXTRAXCTED_GRAPHS_DIR_NAME = Path("fx") / "extracted"
```
[the remark this answers is not in the document]
Comment (daniil-lyakhov) [reply tests/torch/fx/test_model_transformer.py]: Done

File: tests/torch/fx/test_model_transformer.py [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+@pytest.mark.parametrize("test_case", MODEL_EXTRACTION_CASES)
+def test_model_extraction(test_case: ModelExtractionTestCase):
+    with torch.no_grad():
+        with disable_patching():
+            captured_model = capture_pre_autograd_graph(test_case.model(), (torch.ones(test_case.input_shape),))
+    layout = TransformationLayout()
+    layout.register(test_case.command)
+    extracted_model = FXModelTransformer(captured_model).transform(layout)
+    nncf_graph = GraphConverter.create_nncf_graph(extracted_model)
+    check_graph(nncf_graph, f"{test_case.model.__name__}.dot", EXTRAXCTED_GRAPHS_DIR_NAME)
```
[the remark this answers is not in the document]
Comment (daniil-lyakhov) [reply tests/torch/fx/test_model_transformer.py]: Done

File: nncf/experimental/torch/fx/model_transformer.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (daniil-lyakhov) [reply nncf/experimental/torch/fx/model_transformer.py]: Let me know if current version of the comment is not satisfactory

File: tests/torch/test_models/synthetic.py (hunk 2 above)
Comment (daniil-lyakhov) [reply tests/torch/test_models/synthetic.py]: Done

**Review by AlexanderDokuchaev — APPROVED** [review]

**Pull request merged**